scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -521,158 +520,178 @@
                         (map job->usage)
                         (reduce (partial merge-with +)))))))
 
+(defn category->pending-jobs->category->considerable-jobs
+  "Limit the pending jobs to considerable jobs based on usage and quota.
+   Further limit the considerable jobs to a maximum of num-considerable jobs."
+  [db category->pending-jobs user->quota user->usage num-considerable]
+  (log/debug "There are" (apply + (map count category->pending-jobs)) "pending jobs")
+  (log/debug "pending-jobs:" category->pending-jobs)
+  (let [filter-considerable-jobs (fn filter-considerable-jobs [jobs]
+                                   (->> jobs
+                                        (map #(d/entity db (:db/id %)))
+                                        (filter-based-on-quota user->quota user->usage)
+                                        (filter (fn [job]
+                                                  (util/job-allowed-to-start? db job)))
+                                        (take num-considerable)))
+        category->considerable-jobs (->> category->pending-jobs
+                                         (map-vals filter-considerable-jobs))]
+    (log/debug "We'll consider scheduling" (map (fn [[k v]] [k (count v)]) category->considerable-jobs)
+               "of those pending jobs (limited to " num-considerable " due to backdown)")
+    category->considerable-jobs))
+
+(defn extract-matched-job-uuids
```

Comment:
Will do. The beauty is now that we have unit tests, easy to validate if something still works the same :)